sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3957,3 +3957,198 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
 
   override def prettyName: String = "array_except"
 }
+
+/**
+ * Checks if the array (left) has the array (right)
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(array1, array2) - Returns true if the array1 contains the array2.",
```

Comment:
This `array_contains_array` is not the same like others, for example the presto's https://github.com/prestosql/presto/pull/5593.

Looks at the implementation here,  this `array_contains_array` treats array as set and check if the intersect of two sets is the same as the first set.

I'm not sure if there is a good definition for array contains array? For example, how we treat element order? duplicates?

It is not good if we need to fix these issues in the future for this function and add legacy configs for behavior change. It is better to define the function clearly when we add an expression.

Reply:
> This `array_contains_array` is not the same like others, for example the presto's [prestosql/presto#5593](https://github.com/prestosql/presto/pull/5593).
> 
> Looks at the implementation here, this `array_contains_array` treats array as set and check if the intersect of two sets is the same as the first set.
> 
> I'm not sure if there is a good definition for array contains array? For example, how we treat element order? duplicates?
> 
> It is not good if we need to fix these issues in the future for this function and add legacy configs for behavior change. It is better to define the function clearly when we add an expression.

Presto did same thing as what  I did in this pr. but change to current  way. So hope a discussion about what's the final thing spark should do. An d I will change the code.

also  cc @maropu @kiszk @HyukjinKwon

Reply:
First, I will update current pr desc and code comment to make it more clear

Reply:
`array_contains_array` is a weird name to me. `contains_subarray` from Presto and `hasAll` from Clickhouse sound better.

How about we go with `contains_subarray`?

Reply:
> How about we go with `contains_subarray`?

To me, this weekly implies that the function somehow takes into account ordering and / or cardinality of the values. However, at the moment, implementations seems to work like set(right) ⊆ set(left). So maybe `is_subset` would be appropriate, if that's the intended behavior?

Reply:
I prefer `hasAll`.  It doesn't look ambiguous.
`contains_subarray` have some meaning of order. `is_subset` show intended behavior but we put array params.

WDYT?

Reply:
We don't really have set type in spark, so `is_subset` is also weird...

Reply:
> We don't really have set type in spark, so `is_subset` is also weird...

I think so too.. maybe that's the reason why clickhouse call it `hasAll`..

Reply:
`hasAll` (or rather `has_all`, to keep things consitent) sounds good :)